Sofa/Component/Constraint/Lagrangian/Model/src/sofa/component/constraint/lagrangian/model/FixedLagrangianConstraint.inl

```diff
@@ -0,0 +1,102 @@
+/******************************************************************************
+*                 SOFA, Simulation Open-Framework Architecture                *
+*                    (c) 2006 INRIA, USTL, UJF, CNRS, MGH                     *
+*                                                                             *
+* This program is free software; you can redistribute it and/or modify it     *
+* under the terms of the GNU Lesser General Public License as published by    *
+* the Free Software Foundation; either version 2.1 of the License, or (at     *
+* your option) any later version.                                             *
+*                                                                             *
+* This program is distributed in the hope that it will be useful, but WITHOUT *
+* ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or       *
+* FITNESS FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License *
+* for more details.                                                           *
+*                                                                             *
+* You should have received a copy of the GNU Lesser General Public License    *
+* along with this program. If not, see <http://www.gnu.org/licenses/>.        *
+*******************************************************************************
+* Authors: The SOFA Team and external contributors (see Authors.txt)          *
+*                                                                             *
+* Contact information: contact@sofa-framework.org                             *
+******************************************************************************/
+#pragma once
+#include <sofa/component/constraint/lagrangian/model/FixedLagrangianConstraint.h>
+#include <sofa/core/visual/VisualParams.h>
+
+#include <sofa/type/Vec.h>
+
+namespace sofa::component::constraint::lagrangian::model
+{
+
+template<class DataTypes>
+FixedLagrangianConstraint<DataTypes>::FixedLagrangianConstraint(MechanicalState* object)
+    : Inherit(object)
+    , d_indices(initData(&d_indices,  "indices", "Indices of points to fix"))
+    , d_fixAll(initData(&d_fixAll, false,  "fixAll", "If true, fix all points"))
+{}
+
+template<class DataTypes>
+void FixedLagrangianConstraint<DataTypes>::init()
+{
+    if(!this->mstate.get())
+    {
+        this->mstate.setPath("@.");
+    }
+
+    if(this->mstate.get())
+    {
+        msg_warning(this)<<"Data \'mstate\' not specified, defaulting to the mechanical state of the current context";
+        this->d_componentState.setValue(sofa::core::objectmodel::ComponentState::Valid);
+    }
+    else
+    {
+        msg_error(this)<<"Data \'mstate\' not specified, unable to find a compatible mechanical state in the current context";
+        this->d_componentState.setValue(sofa::core::objectmodel::ComponentState::Invalid);
+    }
+}
+
+
+// Vec3D specialization
```

Comment:
Is it really a Vec3D specialization?

Reply:
Is it a real question ? :D thank's for the catch !